dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java

```diff
@@ -54,102 +59,27 @@ public String getVersion(boolean withBuildNumber) {
-                    InputStream mavenPropertiesInputStream = null;
-                    String mavenPropertiesFilePath;
-                    Properties mavenProperties = new Properties();
-
-
-                    filePath = filePath.replaceFirst("/[^/]*$", "/");
-                    // Using a relative path, find the location of the maven pom.properties file. 
-                    // First, try to look for it in the app-level META-INF. This will only be 
-                    // available if it's a war file deployment: 
-                    mavenPropertiesFilePath = filePath.concat("../../../META-INF/maven/edu.harvard.iq/dataverse-webapp/pom.properties");
-
-                    try {
-                        mavenPropertiesInputStream = new FileInputStream(mavenPropertiesFilePath);
-                    } catch (IOException ioex) {
-                        // OK, let's hope this is a local dev. build. 
-                        // In that case the properties file should be available in 
-                        // the maven-archiver directory: 
-
-                        mavenPropertiesFilePath = filePath.concat("../../../../maven-archiver/pom.properties");
-
-                        // try again: 
-
-                        try {
-                            mavenPropertiesInputStream = new FileInputStream(mavenPropertiesFilePath);
-                        } catch (IOException ioex2) {
-                            logger.warning("Failed to find and/or open for reading the pom.properties file.");
-                            mavenPropertiesInputStream = null;
-                        }
-                    }
-
-                    if (mavenPropertiesInputStream != null) {
-                        try {
-                            mavenProperties.load(mavenPropertiesInputStream);
-                            appVersionString = mavenProperties.getProperty("version");
-                        } catch (IOException ioex) {
-                            logger.warning("caught IOException trying to read and parse the pom properties file.");
-                        } finally {
-                            IOUtils.closeQuietly(mavenPropertiesInputStream);
-                        }
-                    }
-
-                } else {
-                    logger.warning("Null file path representation of the location of persistence.xml in the webapp root directory!");
-                }
+            // We'll rely on Maven placing the version number into the
+            // version.properties file using resource filtering
+            
+            Try<String> appVersionTry = Try.withResources(() -> getClass().getResourceAsStream(VERSION_PROPERTIES_CLASSPATH))
+                    .of(is -> {
+                        Properties properties = new Properties();
+                        properties.load(is);
+                        return properties;
+                    })
+                    .map(p -> p.getProperty(VERSION_PROPERTIES_KEY));
+            
```

Comment:
I wonder if we can somehow get rid of the following ```if```?
Maybe we could create Tuple of {result, logging function}, so after line 71 (we would have also to change the type of variable we're assigning that result to) we'd have sth like:
```java
.map(v -> StringUtils.equals(v, VERSION_PLACEHOLDER)
    ? Tuple.of(VERSION_FALLBACK, () -> logger.warning(…))
    : Tuple.of(v, () -> {})
.getOrElse(Tuple.of(VERSION_FALLBACK, () -> logger.warning(…));

// call result._2
return result._1;
```
The only problem of this approach could be proper handling of generic parameters.